src/test/java/com/commercetools/sync/commons/utils/completablefutureutils/MapStreamValuesToFutureOfCompletedValuesTest.java

```diff
@@ -0,0 +1,211 @@
+package com.commercetools.sync.commons.utils.completablefutureutils;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+import static com.commercetools.sync.commons.utils.CompletableFutureUtils.mapValuesToFutureOfCompletedValues;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptySet;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class MapStreamValuesToFutureOfCompletedValuesTest {
+    /**
+     * Stream to List : empty values.
+     * Stream to Set : empty values.
+     **/
+
+    @Test
+    public void empty_StreamToList_ReturnsFutureOfEmptyList() {
```

Comment:
I thought that the test method name should always beginn with the name of the method,which is tested

Reply:
True, but in this case, unit testing the class `CompleteableFutureUtils` in one test class would make it unreadable due to the vast permutations of inputs/outputs to the methods. So every method now has it's own unit test class that has inside all the different input/output combinations. That's why I followed the following approach:

class under test -> package name in the unit tests `/completablefutueutils/`
method under test -> unit test class name (in this case `MapStreamValuesToFutureOfCompletedValuesTest`)
method convention followed -> **`StateUnderTest1_StateUnderTest2_ExpectedBehavior`**

Reply:
thanks for explenation